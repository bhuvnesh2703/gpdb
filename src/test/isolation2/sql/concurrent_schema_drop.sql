-- Test that dropping a new empty schema while concurrently creating
-- objects in that schema blocks the DROP SCHEMA. Each DROP SCHEMA in
-- this test should block trying to obtain AccessExclusiveLock on its
-- corresponding pg_namespace entry.

CREATE SCHEMA concurrent_schema_drop1;
CREATE SCHEMA concurrent_schema_drop2;
CREATE SCHEMA concurrent_schema_drop3;
CREATE SCHEMA concurrent_schema_drop4;
CREATE SCHEMA concurrent_schema_drop5;
CREATE SCHEMA concurrent_schema_drop6;

-- Test on CREATE TABLE
1: BEGIN;
1: CREATE TABLE concurrent_schema_drop1.concurrent_schema_drop1_table(a int);
2&: DROP SCHEMA concurrent_schema_drop1;
1: COMMIT;
2<:
1: SELECT * FROM concurrent_schema_drop1.concurrent_schema_drop1_table;

-- Test on CREATE VIEW
1: BEGIN;
1: CREATE VIEW concurrent_schema_drop4.concurrent_schema_drop4_view AS SELECT 1;
2&: DROP SCHEMA concurrent_schema_drop4;
1: COMMIT;
2<:
1: SELECT * FROM concurrent_schema_drop4.concurrent_schema_drop4_view;

-- Test on ALTER TABLE .. SET SCHEMA
1: CREATE TABLE concurrent_schema_drop6_table(a int);
1: BEGIN;
1: ALTER TABLE concurrent_schema_drop6_table SET SCHEMA concurrent_schema_drop6;
2&: DROP SCHEMA concurrent_schema_drop6;
1: COMMIT;
2<:
1: SELECT * FROM concurrent_schema_drop6.concurrent_schema_drop6_table;
