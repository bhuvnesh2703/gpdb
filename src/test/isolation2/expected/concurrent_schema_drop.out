-- Test that dropping a new empty schema while concurrently creating
-- objects in that schema blocks the DROP SCHEMA. Each DROP SCHEMA in
-- this test should block trying to obtain AccessExclusiveLock on its
-- corresponding pg_namespace entry.

CREATE SCHEMA concurrent_schema_drop1;
CREATE
CREATE SCHEMA concurrent_schema_drop2;
CREATE
CREATE SCHEMA concurrent_schema_drop3;
CREATE
CREATE SCHEMA concurrent_schema_drop4;
CREATE
CREATE SCHEMA concurrent_schema_drop5;
CREATE
CREATE SCHEMA concurrent_schema_drop6;
CREATE

-- Test on CREATE TABLE
1: BEGIN;
BEGIN
1: CREATE TABLE concurrent_schema_drop1.concurrent_schema_drop1_table(a int);
CREATE
2&: DROP SCHEMA concurrent_schema_drop1;  <waiting ...>
1: COMMIT;
COMMIT
2<:  <... completed>
ERROR:  cannot drop schema concurrent_schema_drop1 because other objects depend on it
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
1: SELECT * FROM concurrent_schema_drop1.concurrent_schema_drop1_table;
a
-
(0 rows)

-- Test on CREATE VIEW
1: BEGIN;
BEGIN
1: CREATE VIEW concurrent_schema_drop4.concurrent_schema_drop4_view AS SELECT 1;
CREATE
2&: DROP SCHEMA concurrent_schema_drop4;  <waiting ...>
1: COMMIT;
COMMIT
2<:  <... completed>
ERROR:  cannot drop schema concurrent_schema_drop4 because other objects depend on it
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
1: SELECT * FROM concurrent_schema_drop4.concurrent_schema_drop4_view;
?column?
--------
1       
(1 row)

-- Test on ALTER TABLE .. SET SCHEMA
1: CREATE TABLE concurrent_schema_drop6_table(a int);
CREATE
1: BEGIN;
BEGIN
1: ALTER TABLE concurrent_schema_drop6_table SET SCHEMA concurrent_schema_drop6;
ALTER
2&: DROP SCHEMA concurrent_schema_drop6;  <waiting ...>
1: COMMIT;
COMMIT
2<:  <... completed>
ERROR:  cannot drop schema concurrent_schema_drop6 because other objects depend on it
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
1: SELECT * FROM concurrent_schema_drop6.concurrent_schema_drop6_table;
a
-
(0 rows)
